cpp/include/cuspatial/cubic_spline.hpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <cudf/column/column.hpp>
+#include <cudf/table/table.hpp>
+
+namespace cuspatial {
+
+/**
+ * @brief Compute cubic interpolations of a set of points based on their
+ * ids and a coefficient matrix.
+ *
+ * @param[in] query_points column of coordinate values to be interpolated.
+ *
+ * @param[in] spline_ids ids that identift the spline to interpolate each
+ * coordinate into.
+ *
+ * @param[in] offsets int32 column of offset of the source_points.
+ * This is used to calculate which values from the coefficients are
+ * used for each interpolation.
+ *
+ * @param[in] source_points column of the original `t` values used
+ * to compute the coefficients matrix.  These source points are used to
+ * identify which specific spline a given query_point is interpolated with.
+ *
+ * @param[in] coefficients table of spline coefficients produced by
+ * cubicspline_coefficients.
+ *
+ * @return cudf::column `y` coordinates interpolated from `x` and `coefs`.
+**/
+std::unique_ptr<cudf::column> cubicspline_interpolate(
+                                         cudf::column_view const& query_points,
+                                         cudf::column_view const& spline_ids,
+                                         cudf::column_view const& offsets,
+                                         cudf::column_view const& source_points,
+                                         cudf::table_view const& coefficients,
+                                         rmm::mr::device_memory_resource *mr,
```

Comment:
`mr` should default to the default resource:

```suggestion
                                         rmm::mr::device_memory_resource *mr =
                                           rmm::mr::get_default_resource(),
```

Reply:
I didn't have time to add this, yet.

Reply:
It's important to get the APIs right before we merge.